cuegui/cuegui/MenuActions.py

```diff
@@ -1046,30 +1074,64 @@ def retry(self, rpcObjects=None):
                 job.retryFrames(name=names)
                 self._update()
 
-    previewMain_info = ["Preview Main", None, "previewMain"]
+    previewMain_info = ["Preview Main", None, "image"]
 
     # pylint: disable=broad-except
-    def previewMain(self, rpcObjects=None):
+    def previewMain(self, rpcObjects=None, katanaMode=False):
         try:
             job = self._getSource()
             frame = self._getOnlyFrameObjects(rpcObjects)[0]
-            d = cuegui.PreviewWidget.PreviewProcessorDialog(job, frame, False)
-            d.process()
-            d.exec_()
+            layer = None
+            layers = job.getLayers()
+            if len(layers) > 0:
+                for ilayer in layers:
+                    if ilayer.name() == frame.layer():
+                        layer = ilayer
+
+            if layer is not None:
+                outputs = layer.getOutputPaths()
+                if len(outputs) > 0:
+                    frames = self._getOnlyFrameObjects(rpcObjects)
+                    frameSet = FileSequence.FrameSet(','.join(str(f.number()) for f in frames))
+                    cuegui.Utils.previewOutputs([outputs[0]], frameSet=frameSet)
+                else:
+                    katanaMode = True
+            if katanaMode is True:
+                d = cuegui.PreviewWidget.PreviewKatanaProcessorDialog(job, frame, False)
+                d.process()
+                d.exec_()
         except Exception as e:
             QtWidgets.QMessageBox.critical(None, "Preview Error",
                                            "Error displaying preview frames, %s" % e)
 
-    previewAovs_info = ["Preview All", None, "previewAovs"]
+    previewAovs_info = ["Preview All", None, "images"]
 
     # pylint: disable=broad-except
-    def previewAovs(self, rpcObjects=None):
+    def previewAovs(self, rpcObjects=None, katanaMode=False):
         try:
             job = self._getSource()
             frame = self._getOnlyFrameObjects(rpcObjects)[0]
-            d = cuegui.PreviewWidget.PreviewProcessorDialog(job, frame, True)
-            d.process()
-            d.exec_()
+            layer = None
+            layers = job.getLayers()
+            if len(layers) > 0:
+                for ilayer in layers:
+                    if ilayer.name() == frame.layer():
+                        layer = ilayer
```

Comment:
Same suggestion as above, checking for emptiness before a loop is redundant

Reply:
Changed